pennylane_lightning/lightning_gpu/_state_vector.py

```diff
@@ -37,11 +70,308 @@ class LightningGPUStateVector(LightningBaseStateVector):
         device_name(string): state vector device name. Options: ["lightning.gpu"]
     """
 
-    def __init__(self, num_wires, dtype=np.complex128, device_name="lightning.gpu"):
-
-        if device_name != "lightning.gpu":
-            raise DeviceError(f'The device name "{device_name}" is not a valid option.')
+    def __init__(
+        self,
+        num_wires,
+        dtype=np.complex128,
+        device_name="lightning.gpu",
+        mpi_handler=None,
```

Comment:
Do we want to pass `mpi_handler` around?

Reply:
Yes because I tried to keep the MPI parts inside the LGPU class but at some point I had circular import problems. That is why I created an independent class. Also, It will be necessary to pass `mpi_handler` to get the functionality of the MPI methods.